src/rpc/blockchain.cpp

```diff
@@ -286,14 +287,18 @@ static RPCHelpMan waitfornewblock()
     NodeContext& node = EnsureAnyNodeContext(request.context);
     Mining& miner = EnsureMining(node);
 
-    auto block{CHECK_NONFATAL(miner.getTip()).value()};
-    if (IsRPCRunning()) {
-        block = timeout ? miner.waitTipChanged(block.hash, std::chrono::milliseconds(timeout)) : miner.waitTipChanged(block.hash);
-    }
+    std::optional<BlockRef> block{miner.getTip()};
+    // If getTip() returns null, which can happen during startup, pass 0 to
+    // waitTipChanged() so it will wait for any tip hash to be set.
+    uint256 tip_hash{block ? block->hash : uint256{}};
```

Comment:
nit: maybe use `uint256::ZERO`, also used in commit b954af72e4d6a83b3ba701b7ee52d97f2af3254b `Have createNewBlock() wait for a tip`:

```suggestion
    uint256 tip_hash{block ? block->hash : uint256::ZERO};
```